src/Nethermind/Nethermind.Network/TrustedNodes/TrustedNodesManager.cs

```diff
@@ -0,0 +1,172 @@
+        private readonly ILogger _logger;
+
+        public TrustedNodesManager(string trustedNodesPath, ILogManager logManager)
+        {
+            _trustedNodesPath = trustedNodesPath.GetApplicationResourcePath();
+            _logger = logManager.GetClassLogger();
+        }
+
+        public IEnumerable<NetworkNode> Nodes => _nodes.Values;
+
+        public async Task InitAsync()
+        {
+            if (!File.Exists(_trustedNodesPath))
+            {
+                if (_logger.IsDebug)
+                {
+                    _logger.Debug($"Trusted nodes file not found at: {_trustedNodesPath}");
+                }
+                return;
+            }
+
+            HashSet<string> lines = new();
+            // Read lines asynchronously
+            await foreach (string line in File.ReadLinesAsync(_trustedNodesPath))
+            {
+                if (!string.IsNullOrWhiteSpace(line))
+                {
+                    lines.Add(line);
+                }
+            }
+
+            if (_logger.IsInfo)
+            {
+                _logger.Info($"Loaded {lines.Count} trusted nodes from: {Path.GetFullPath(_trustedNodesPath)}");
+            }
+
+            if (lines.Any() && _logger.IsDebug)
+            {
+                _logger.Debug("Trusted nodes:\n" + string.Join(Environment.NewLine, lines));
+            }
+
+            // Parse each line into a NetworkNode
+            List<NetworkNode> networkNodes = new();
+            foreach (string line in lines)
+            {
+                try
+                {
+                    networkNodes.Add(new NetworkNode(line));
+                }
+                catch (Exception ex) when (ex is ArgumentException or SocketException)
+                {
+                    if (_logger.IsError)
+                    {
+                        _logger.Error($"Failed to parse '{line}' as a trusted node.", ex);
+                    }
+                }
+            }
+
+            _nodes = new ConcurrentDictionary<PublicKey, NetworkNode>(
+                networkNodes.ToDictionary(n => n.NodeId, n => n));
+        }
+
+        // ---- INodeSource requirement: IAsyncEnumerable<Node> ----
+        // C# requires 'async' for IAsyncEnumerable yield. We'll add a small 'await Task.Yield()'
+        // to avoid the warning about "no awaits".
+        public async IAsyncEnumerable<Node> DiscoverNodes([EnumeratorCancellation] CancellationToken cancellationToken)
+        {
+            // At least one 'await', so no compiler warnings
+            await Task.Yield();
+
+            foreach (NetworkNode netNode in _nodes.Values)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+                yield return new Node(netNode) { IsTrusted = true };
+            }
+        }
+
+        public async Task<bool> AddAsync(Enode enode, bool updateFile = true)
+        {
+            NetworkNode networkNode = new(enode.ToString());
```

Comment:
Add `NetworkNode` constructor that takes `Enode` as class and not string to avoid double-conversion.